### PR TITLE
Publish ambient types

### PR DESCRIPTION
**Pull request by gitKrystan**
Needed for https://github.com/emberjs/ember-qunit/pull/994
See https://github.com/emberjs/ember.js/issues/20162

### CODE REVIEW

File: test-support/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+export default class TestLoader {
```
Comment (gitKrystan) [inline test-support/index.d.ts:1]: Based on the docs for this library, I'm not sure how much of this should actually be public. All I need for https://github.com/emberjs/ember-qunit/pull/994 is `load`, `moduleLoadFailure`, `addModuleExcludeMatcher` and `addModuleIncludeMatcher`.

Comment (chancancode) [reply test-support/index.d.ts:1]: Should we comment out the rest and see if anyone asks for them?

Comment (chancancode) [reply test-support/index.d.ts:1]: I guess at the end of the day this whole package is probably pretty "private" anyway 🤷🏼‍♂️

Comment (chancancode) [reply test-support/index.d.ts:1]: Ok so looking at the implementation, I would guess that `shouldLoadModule`, `listModules`, `listTestModules`, `require`, `unsee` and `moduleLoadFailure` are probably meant to be private helpers.

Comment (gitKrystan) [reply test-support/index.d.ts:1]: `ember-qunit` does `new TestLoader().load()`

Comment (chancancode) [reply test-support/index.d.ts:1]: OTOH, the readme says....

>   // optionally override TestLoader.prototype.shouldLoadModule

🤷🏼 🤷🏼 🤷🏼 

I guess you can take all of that information and decide what to do with them as you see fit, this is just an "internal" support package and I don't think these things were thought through very rigorously at the time.

Comment (chancancode) [reply test-support/index.d.ts:1]: > `ember-qunit` does `new TestLoader().loadModules()`

It probably should have just called `load()` which does exactly that, but oh well

Comment (gitKrystan) [reply test-support/index.d.ts:1]: lol true

Comment (gitKrystan) [reply test-support/index.d.ts:1]: README says "optionally override TestLoader.prototype.shouldLoadModule" so I am going to make that public also

Comment (gitKrystan) [reply test-support/index.d.ts:1]: I am working on ember-qunit anyway, so I just made that call the static `load` method and made `loadModules` private and the constructor protected.

**Review by chancancode — APPROVED** [review]

**Review by chancancode — APPROVED** [review]

**Pull request merged**